includes/discovery/vlans/jetstream.inc.php

```diff
@@ -29,14 +29,34 @@
 // tested on: T1600G-28TS 3.0; T2600G-18TS 2.0;
 //
 // todo: detect LAG ports ??? now parser assume that there is no LAG port
+// 2021-06-07: Added Vlan parsing on LAG ports
 //
 
 if (! function_exists('jetstreamExpand')) {
     function jetstreamExpand($var)
     {
+        unset($result);
+
+        if ($pos = strpos($var, ',LAG')) {         // if input string contains 'LAG'
+            $lag = str_replace('LAG', '', substr($var, $pos + 1)); // get the LAG ports in form of '1,2,4-8'
```

Comment:
This code is really hard to read, could it perhaps be done with a regex?

Reply:
hi @murrant 

sorry, my "native" language is assembler on embedded devices, so i am not familiar with PHP
if you have any suggestion ...
TP-LINK made own way of representing Vlans in snmp
for example:
STRING: "1/0/1-2,1/0/4-6,1/0/25,LAG1-2,LAG4"
as you see, port 1,2,4,5,6,25 are tagged
and LAG1,2,4 are tagged
so, the information about ports and LAG's are in one line
it need to be splitted on two parts, first for ports, second for LAG's
then DASH need to be replaced with corresponding numbers
4-6 -> 4,5,6
same technique for PORTS (left) part of string
and same for LAG's (right) part of string

anyway, if you could / want to rewrite this i will be happy

Reply:
Try using a site like regex101.com  https://regex101.com/r/7utMDz/1

Sample code:
```php
$string = '1/0/1-2,1/0/4-6,1/0/25,LAG1-2,LAG4';
preg_match_all('/LAG(\d+)(?:-(\d+))?/', $string, $matches);
foreach ($matches[1] as $index => $start) {
    $end = $matches[2][$index] ?: $start;
    for ($i = $start; $i <= $end; $i++) {
        dump((int)$i);
    }
}
```

Reply:
hi @murrant 
following your advice , the whole function was rewritten ...

    function jetstreamExpand($var)
    {
        unset($result);

        preg_match_all('/LAG(\d+)(?:-(\d+))?/', $var, $lags);

        foreach ($lags[1] as $index => $start) {
            $end = $lags[2][$index] ?: $start;
            for ($i = $start; $i <= $end; $i++) {
                $result[] = 'LAG' . $i; //need to be in form LAGx
            }
        }

        preg_match_all('/1\/0\/(\d+)(?:-(\d+))?/', $var, $ports);
        foreach ($ports[1] as $index => $start) {
            $end = $ports[2][$index] ?: $start;
            for ($i = $start; $i <= $end; $i++) {
                $result[] = '1/0/' . $i; //need to be in form 1/0/x
            }
        }

        return $result;
    }